### PR TITLE
[24.0] Datasets list anonymous access and history link fixes.

### PR DESCRIPTION
**Pull request by dannon**
Two minor fixes for "Datasets" list/grid, fixes https://github.com/galaxyproject/galaxy/issues/17730

This form also had the same issue as was resolved in https://github.com/galaxyproject/galaxy/pull/17616, so I've standardized the history display to use that component, slightly enhanced to accept optional filters.

Anonymous users are constrained to a single history and all datasets are displayed there, so this grid is disabled (the datasets list/query backend is also not supported in the API -- if we do want to support anon we'd want to adjust that first in a larger fix).

I know this one is slated for an overhaul next cycle, but the endpoint access can be considered then along with other tweaks like thinking about the utility of the dropdown instead of a simpler mechanism for copy-to-history, etc.  These could totally be draggable dataset cards or something, if we reimagine this.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: client/src/entry/analysis/menu.js [hunk 1]
```diff
@@ -105,6 +105,7 @@ export function fetchMenu(options = {}) {
                 {
                     title: _l("Datasets"),
                     url: "/datasets/list_published",
```
Comment (martenson) [inline client/src/entry/analysis/menu.js:107]:
```suggestion
                    url: "/datasets/list",
```

Comment (martenson) [reply client/src/entry/analysis/menu.js:107]: there is no grid for public datasets, this url was added erroneously during the transition under "Data" in masthead

File: client/src/entry/analysis/menu.js [hunk 2]
```diff
@@ -105,6 +105,7 @@ export function fetchMenu(options = {}) {
...
+                    hidden: !Galaxy.user.id,
                 },
```
Comment (martenson) [inline client/src/entry/analysis/menu.js:109]:
```suggestion
```

Comment (martenson) [reply client/src/entry/analysis/menu.js:109]: there is no grid for public datasets, this url was added erroneously during the transition under "Data" in masthead

Comment (mvdbeek) [reply client/src/entry/analysis/menu.js:109]: I think you're right, I'll merge and open a followup

**Pull request merged**